### PR TITLE
T5756: L2TP RADIUS backup and weight settings

### PR DESCRIPTION
**Pull request by HollyGurza**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5756

## Smoketest result

```
vyos@vyos:~$ python3 /usr//libexec/vyos/tests/smoke/cli/test_vpn_l2tp.py 
test_accel_ipv4_pool (__main__.TestVPNL2TPServer.test_accel_ipv4_pool) ... ok
test_accel_ipv6_pool (__main__.TestVPNL2TPServer.test_accel_ipv6_pool) ... 
WARNING: IPv4 Server requires gateway-address to be configured!


WARNING: 'default-ipv6-pool' is not defined

ok
test_accel_limits (__main__.TestVPNL2TPServer.test_accel_limits) ... ok
test_accel_local_authentication (__main__.TestVPNL2TPServer.test_accel_local_authentication) ... 
User "test" has rate-limit configured for only one direction but both
upload and download must be given!

ok
test_accel_name_servers (__main__.TestVPNL2TPServer.test_accel_name_servers) ... ok
test_accel_next_pool (__main__.TestVPNL2TPServer.test_accel_next_pool) ... 
WARNING: 'default-pool' is not defined

ok
test_accel_ppp_options (__main__.TestVPNL2TPServer.test_accel_ppp_options) ... ok
test_accel_radius_authentication (__main__.TestVPNL2TPServer.test_accel_radius_authentication) ... ok
test_accel_shaper (__main__.TestVPNL2TPServer.test_accel_shaper) ... ok
test_accel_snmp (__main__.TestVPNL2TPServer.test_accel_snmp) ... ok
test_accel_wins_server (__main__.TestVPNL2TPServer.test_accel_wins_server) ... ok
test_l2tp_radius_server (__main__.TestVPNL2TPServer.test_l2tp_radius_server) ... set vpn l2tp remote-access authentication mode radius
set vpn l2tp remote-access gateway-address 192.0.2.1
set vpn l2tp remote-access client-ip-pool SIMPLE-POOL range 192.0.2.0/24
set vpn l2tp remote-access default-pool SIMPLE-POOL
set vpn l2tp remote-access authentication radius server 192.0.2.22 key secretVyOS
set vpn l2tp remote-access authentication radius server 192.0.2.22 weight 10
set vpn l2tp remote-access authentication radius server 192.0.2.22 backup
del vpn l2tp remote-access
ok
test_l2tp_server_authentication_protocols (__main__.TestVPNL2TPServer.test_l2tp_server_authentication_protocols) ... ok
test_vpn_l2tp_dependence_ipsec_swanctl (__main__.TestVPNL2TPServer.test_vpn_l2tp_dependence_ipsec_swanctl) ... 
WARNING: 'default-pool' is not defined

ok

----------------------------------------------------------------------
Ran 14 tests in 58.672s

OK

```
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/include/accel-ppp/radius-additions.xml.i [hunk 1]
```diff
@@ -57,6 +57,24 @@
           </properties>
           <defaultValue>0</defaultValue>
         </leafNode>
+        <leafNode name="weight">
+          <properties>
```
Comment (c-po) [inline interface-definitions/include/accel-ppp/radius-additions.xml.i]: I think we have something like this (priority) for RADIUS system login servers. We should keep a consistent CLI

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/include/accel-ppp/radius-additions.xml.i (hunk 1 above)
Comment (HollyGurza) [reply interface-definitions/include/accel-ppp/radius-additions.xml.i]: i agree
fixed, now `priority` in CLI but `weight` in config.

And what about the default value? in the RADIUS system login servers we have the default as 255 but in accel-ppp documentation the default is 1 and our CLI don't set this value for accel-ppp by default.

i think naming support will be enough

Comment (c-po) [reply interface-definitions/include/accel-ppp/radius-additions.xml.i]: Yes. Please use consistent naming. The defaultValue can be overwritten for login and accel.ppp

File: interface-definitions/include/accel-ppp/radius-additions.xml.i [hunk 2]
```diff
@@ -57,6 +57,24 @@
           </properties>
           <defaultValue>0</defaultValue>
         </leafNode>
+        <leafNode name="priority">
```
Comment (c-po) [inline interface-definitions/include/accel-ppp/radius-additions.xml.i]: Please use a common `#include` for the priority node.

Move https://github.com/vyos/vyos-1x/blob/current/interface-definitions/system_login.xml.in#L205C1-L217C30 to `include/radius-priority.xml.i` and include the snipped in both places. Only set defaultValue 255 for system-login

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/include/accel-ppp/radius-additions.xml.i (hunk 2 above)
Comment (HollyGurza) [reply interface-definitions/include/accel-ppp/radius-additions.xml.i]: done

Comment (HollyGurza) [reply interface-definitions/include/accel-ppp/radius-additions.xml.i]:
```
vyos@vyos# set vpn l2tp remote-access authentication radius server 192.168.0.1 [tab]
Possible completions:
   acct-port            Accounting port (default: 1813)
   backup               Use backup server if other servers are not available
   disable              Disable instance
   disable-accounting   Disable accounting
   fail-time            Mark server unavailable for <n> seconds on failure (default: 0)
   key                  Shared secret key
   port                 Port number used by connection (default: 1812)
   priority             Server priority
```

```
vyos@vyos# set system login radius server 192.168.0.1 [tab]
Possible completions:
   disable              Disable instance
   key                  Shared secret key
   port                 Port number used by connection (default: 1812)
   priority             Server priority (default: 255)
   timeout              Session timeout (default: 2)
```

**Review by c-po — APPROVED** [review]

**Comment by c-po** [comment]
@MergifyIo backport sagitta

**Pull request merged**